packages/unified-selection/src/unified-selection/SelectionScope.ts

```diff
@@ -0,0 +1,279 @@
+                        JOIN BisCore.Element e ON e.ECInstanceId = etae.ECInstanceId
+                        JOIN BisCore.Element pe ON pe.ECInstanceId = e.Parent.Id
+                    ),
+                    Element3dTopAssemblyRelatedFunctionalElements (ECInstanceId, ClassName) AS (
+                      SELECT COALESCE(r.TargetECInstanceId, etae.ECInstanceId), ec_classname(COALESCE(r.TargetECClassId, etae.ECClassId), 's.c')
+                      FROM Element3dTopAssemblyElements etae
+                        JOIN BisCore.Element e ON e.ECInstanceId = etae.ECInstanceId
+                        LEFT JOIN Functional.PhysicalElementFulfillsFunction r ON r.SourceECInstanceId = etae.ECInstanceId
+                      WHERE e.Parent.Id IS NULL
+                    )
+                    SELECT * FROM Element2dTopAssemblyRelatedFunctionalElements
+                    UNION
+                    SELECT * FROM Element3dTopAssemblyRelatedFunctionalElements;`;
+
+    yield* executeQuery(queryExecutor, query, bindings);
+  }
+
+  function getNearestFunctionalElementQuery(ids: string[], bindings: ECSqlBinding[]): string {
+    return `Element2dNearestFunctionalElement (OriginalECInstanceId, OriginalECClassId, ECInstanceId, ECClassId, ParentId) AS (
+              SELECT ECInstanceId, ECClassId, ECInstanceId, ECClassId, Parent.Id FROM BisCore.GeometricElement2d WHERE ${formIdBindings("ECInstanceId", ids, bindings)}
+              UNION ALL
+              SELECT e.OriginalECInstanceId, e.OriginalECClassId, COALESCE(r.TargetECInstanceId, ge.ECInstanceId), COALESCE(r.TargetECClassId, ge.ECClassId), ge.Parent.Id
+              FROM Element2dNearestFunctionalElement e
+                LEFT JOIN BisCore.Element ge ON ge.ECInstanceId = e.ParentId
+                LEFT JOIN Functional.DrawingGraphicRepresentsFunctionalElement r ON r.SourceECInstanceId = e.ECInstanceId
+              WHERE e.ECClassId IS NOT (Functional.FunctionalElement) AND (e.ParentId IS NOT NULL OR r.TargetECInstanceId IS NOT NULL)
+            ),
+            FoundNearestFunctionalElements (OriginalECInstanceId, ECInstanceId, ECClassId) AS (
+              SELECT OriginalECInstanceId, ECInstanceId, ECClassId
+              FROM Element2dNearestFunctionalElement
+              WHERE ECClassId IS (Functional.FunctionalElement)
+            ),
+            ElementsWithoutFunctionalElement (ECInstanceId, ECClassId) AS (
+              SELECT e.OriginalECInstanceId, OriginalECClassId
+              FROM Element2dNearestFunctionalElement e
+                LEFT JOIN FoundNearestFunctionalElements f ON f.OriginalECInstanceId = e.OriginalECInstanceId
+              WHERE e.ParentId IS NULL AND f.ECInstanceId IS NULL
+            ),
+            MergedElements (ECInstanceId, ECClassId) AS (
+              SELECT ECInstanceId, ECClassId FROM FoundNearestFunctionalElements
+              UNION
+              SELECT ECInstanceId, ECClassId FROM ElementsWithoutFunctionalElement
+            ),`;
```

Comment:
I think the last comma doesn't belong here.

Reply:
Since the functional scopes merged, I merged this part into the main query.